terminus-ui/logo/src/logo.component.ts

```diff
@@ -0,0 +1,159 @@
+import { FULL_GRADIENT_LOGO } from './logo-types/full_gradient';
+import { FULL_SOLID_LOGO } from './logo-types/full_solid';
+import { MARK_GRADIENT_LOGO } from './logo-types/mark_gradient';
+import { MARK_SOLID_LOGO } from './logo-types/mark_solid';
+
+
+/**
+ * Currently supported logo types
+ */
+export type TS_LOGO_TYPE
+  = 'full-account-hub'
+  | 'full-gradient'
+  | 'full-solid'
+  | 'mark-gradient'
+  | 'mark-solid'
+;
+
+
+/**
+ * An array of supported logo types.
+ */
+export const TS_LOGO_TYPES: TS_LOGO_TYPE[] = [
+  'full-account-hub',
+  'full-gradient',
+  'full-solid',
+  'mark-gradient',
+  'mark-solid',
+];
+
+/**
+ * Currently supported logo colors
+ */
+export type TS_LOGO_COLOR
+  = 'white'
+  | 'black'
+  | 'gray'
+;
+
+
+/**
+ * An array of supported logo colors.
+ */
+export const TS_LOGO_COLORS: TS_LOGO_COLOR[] = [
+  'white',
+  'black',
+  'gray',
+];
+
+
+/**
+ * This is the logo UI Component
+ *
+ * #### QA CSS CLASSES
+ * - `qa-label`: Placed on the primary container
+ *
+ * @example
+ * <ts-logo type="full-gradient"></ts-logo>
+ * <ts-logo type="solid-mark" logoColor="gray"></ts-logo>
+ *
+ * <example-url>https://getterminus.github.io/ui-demos-master/components/logo</example-url>
+ */
+@Component({
+  selector: 'ts-logo',
+  templateUrl: './logo.component.html',
+  styleUrls: ['./logo.component.scss'],
+  host: {
+    class: 'ts-logo',
+    '[class.ts-logo--black]': 'logoColor === "black"',
+    '[class.ts-logo--gray]': 'logoColor === "gray"',
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  exportAs: 'tsLogo',
+})
+
+export class TsLogoComponent implements OnInit {
+  /**
+   * Name of the logo types
+   */
+  @Input()
+  public set type(value: TS_LOGO_TYPE | undefined) {
+    // If an unsupported value is passed in
+    if (value && TS_LOGO_TYPES.indexOf(value) < 0 && isDevMode()) {
+      console.warn(`TsLogoComponent: "${value}" is not a supported logo type. ` +
+      `See TS_LOGO_TYPE for available options.`);
+      return;
+    }
+    this._type = value;
+  }
+  public get type(): TS_LOGO_TYPE | undefined {
+    return this._type;
+  }
+  private _type: TS_LOGO_TYPE | undefined = 'full-gradient';
+
+
+  /**
+   * Name of the logo colors
+   */
+  @Input()
+  public set logoColor(value: TS_LOGO_COLOR | undefined) {
+    // If an unsupported value is passed in
+    if (value && TS_LOGO_COLORS.indexOf(value) < 0 && isDevMode()) {
+      console.warn(`TsLogoComponent: "${value}" is not a supported logo color. ` +
+      `See TS_LOGO_COLOR for available options.`);
+      return;
+    }
+    this._logoColor = value;
+  }
+  public get logoColor(): TS_LOGO_COLOR | undefined {
+    return this._logoColor;
+  }
+  private _logoColor: TS_LOGO_COLOR | undefined = 'white';
+
+
+  public svg: SafeHtml = ``;
+
+
+  constructor(
+    private domSanitizer: DomSanitizer,
+  ) {}
+
+  ngOnInit() {
+    switch (this.type) {
```

Comment:
Let's add `public` before this and `:void` as the return type.

I'd also suggest moving this logic into another method and then calling that method from `ngOnInit`. In general we should be trying to keep logic out of lifecycle events.